bundle-workflow/src/test_workflow/test_result/test_result_component.py

```diff
@@ -0,0 +1,23 @@
+from sortedcontainers import SortedDict  # type: ignore
+
+
+class TestResultsComponent(SortedDict):
+    def __init__(self):
+        super(TestResultsComponent, self).__init__()
+
+    def __append__(self, result):
+        self.__setitem__(result.config, result)
+
+    def append(self, result):
+        self.__append__(result)
+
+    def log(self):
+        for result in self.values():
+            result.log(str(result))
+
+    def test_status(self):
```

Comment:
This method name is confusing, a status of True/False is odd. Call this `failed` and make it a `@property`.

You can also return early from the loop as soon as you find a failed test. Try using the `next()` iterator.